### ISSUE TITLE
Encoding issue while fetching rank list

### ISSUE THREAD

**Issue by Xdynix** [upbit/pixivpy#26]
Hi, I've found that the json result of `ranking_all()` is bad encoding(at least in my environment), there may occurs things like:
```
  'tags': ['Fate/GrandOrder',
           'Fate',
           'ĺŁŤĺ\x89Ł',
           'ĺŁŤă\x81\x90ă\x81\xa0',
           'Fate/GO500usersĺ\x85Ľă\x82\x8a'],
```

And I also found that if I edit the code in `api.BasePixivAPI.requests_call()` to this:
```
    def requests_call(self, method, url, headers={}, params=None, data=None, stream=False):
        """ requests http/https call for Pixiv API """
        try:
            if (method == 'GET'):
                r = requests.get(url, params=params, headers=headers, stream=stream, **self.requests_kwargs)
            elif (method == 'POST'):
                r = requests.post(url, params=params, data=data, headers=headers, stream=stream, **self.requests_kwargs)
            elif (method == 'DELETE'):
                r = requests.delete(url, params=params, data=data, headers=headers, stream=stream, **self.requests_kwargs)
            r.encoding = 'utf-8'  # Manually set the encoding
            return r
        except Exception as e:
            raise PixivError('requests %s %s error: %s' % (method, url, e))

        raise PixivError('Unknow method: %s' % method)
```
Everything will be fine.

I hope this helps you.

**Comment by upbit** [comment]
Great work! Now I realized that Public-API does not specify the encoding in response headers. After manually set the encoding to UTF-8, issues #11 #18 have been fixed.

Because the App-API (6.x API) has no encoding BUG, I fixed it in `PixivAPI.auth_requests_call()`:

~~~py
class PixivAPI(BasePixivAPI):

    def auth_requests_call(self, method, url, headers={}, params=None, data=None):
        ...
        r = self.requests_call(method, url, headers, params, data)
        r.encoding = 'utf-8'  # Manually set the encoding due to #18,#26
        return r
~~~

~~~py
    api = PixivAPI()
    api.login(_USERNAME, _PASSWORD)

    json_result = api.works(56583142)
    print(json_result)
    illust = json_result.response[0]
    print(">>> %s, tags: %s" % (illust.caption, illust.tags))
~~~

Thank you very much for pointing that out!

**Comment by upbit** [comment]
Fixed in v3.2.2
